src/pages/products/index.tsx

```diff
@@ -14,67 +13,99 @@ import {
   listWrapper,
   opacityBox,
   productsWrapper,
-  sampleImage,
   span,
   statusWrapper,
   textBox,
   title,
   wrapper,
+  dropdownOverlayStyle,
+  dropdownMenuStyle,
+  menuItemStyle,
+  soldOutOverlayStyle,
 } from './index.styles';
 import { useState } from 'react';
+import { useGetProductsQuery } from '@/queries/products';
+import { useLocation } from 'react-router-dom';
+
+interface ProductType {
+  categoryId: number;
+  id: number;
+  description: string;
+  name: string;
+  images: string[];
+  isLiked: boolean;
+  price: number;
+  sellerId: string;
+  status: string;
+}
 
 export default function Products() {
+  const location = useLocation();
+  const { id, name } = location.state || {}; // 전달된 Props 데이터 추출
+  const [status, setStatus] = useState<number>();
+  const category = id;
+
+  const { data } = useGetProductsQuery({ category, status });
   const [isOpen, setIsOpen] = useState(false); // 드롭다운 열림/닫힘 상태
```

Comment:
`useState` 사용할 때에도 type을 지정하면 좋습니다!
```suggestion
  const [isOpen, setIsOpen] = useState<boolean>(false); // 드롭다운 열림/닫힘 상태
```